### ISSUE TITLE
FunctionComment.WrongStyle on unrelated comment

### ISSUE THREAD

**Issue by glen-84** [squizlabs/PHP_CodeSniffer#1961]
```php
    // (other code here)

    // --- Utility methods --- //

    protected function stringsToIntegers(array $strings)
    {
        return array_map(function ($string) {
            return (int)$string;
        }, $strings);
    }
```

Gives:

```
 87 | ERROR | You must use "/**" style comments for a function
    |       | comment (Squiz.Commenting.FunctionComment.WrongStyle)
```

With the empty line before the method, it shouldn't be associated with that comment in my opinion.

In fact, without that blank line, you would get a `There must be no blank lines after the function comment` message anyway (`Squiz.Commenting.FunctionComment.SpacingAfter`).

There's no way to create a "sectioning" comment.

**Comment by gsherwood** [comment]
There is no way for the sniff to know that the comment that you've placed before the function is not actually a function comment. You can certainly create sectioning comments, but you also need to comment your functions when you do.

If you don't want to comment your functions, I'd suggest removing the `Squiz.Commenting.FunctionComment` sniff from your standard, or just muting the `Squiz.Commenting.FunctionComment.WrongStyle` error message if you just want this specific error gone.

Is that an ok solution?

**Comment by glen-84** [comment]
Does anyone put blank lines between doc comments and functions?

```php
/**
 * My description
 */

public function test()
{
}
```

As I said, that would actually result in `Squiz.Commenting.FunctionComment.SpacingAfter`.

If you don't want to change this, then I'll have to just work around it.

**Comment by gsherwood** [comment]
> Does anyone put blank lines between doc comments and functions?

Maybe some people do, but that's not the standard that this sniff is trying to enforce. It is specially requiring no spaces there. If you want to allow blank lines there, you can set the severity of `Squiz.Commenting.FunctionComment.SpacingAfter` to `0` in your ruleset.xml file, but there is no sniff that will enforce a single space between the comment and the function declaration because nobody has written one yet.

**Comment by glen-84** [comment]
> there is no sniff that will enforce a single space between the comment and the function declaration

I don't want that. I was just saying that if there *is* a space (one or more blank lines), then it's probably not intended to be a function comment (*especially* if it's not even a doc comment, as in my initial example).

Anyway, I'll work around the problem and close this issue.

Thanks Greg.
